src/Application.vala

```diff
@@ -282,10 +283,17 @@ public class AppCenter.App : Gtk.Application {
         switch (operation) {
             case AppCenterCore.Package.State.INSTALLING:
                 if (error == null) {
+                    last_installed_package = package;
+
                     // Check if window is focused
                     if (main_window != null) {
                         var win = main_window.get_window ();
                         if (win != null && (win.get_state () & Gdk.WindowState.FOCUSED) != 0) {
+                            var toast = main_window.toast;
+                            toast.title = _("“%s” has been installed").printf (package.get_name ());
+
+                            toast.send_notification ();
```

Comment:
It would be better to handle sending the toast and saving the package reference inside MainWindow, instead of trying to expose `last_installed_package` in Application and then accessing it back in MainWindow. 

Simply declare a `private AppCenterCore.Package? last_installed_package` member and a `public void send_installed_toast (AppCenterCore.Package package)` method in MainWindow and manage sending the toast there instead.

Reply:
I now also need `selected_package` in `AppCenter.App` to check if a `Toast` should be sent.

Reply:
I mean sure, but that didn't resolve my issue. The toast is still half managed in Application and MainWindow at the same time. And there's also no need to store `selected_package` globaly inside Application. Notice how MainWindow actually calls `show_package` and `return_clicked`  where this property could be set within MainWindow as well.

Reply:
Thanks for your feedback. I have refactored that except for `selected_package`.

The `show_package` call in MainWindow is not the same as in View.

Reply:
I refactored it using a signal.

@donadigo thanks for being persistent 😃 The code looks much cleaner now 👍